File: backend/assemblies/Employee.Performance.Evaluator.Application/RequestsAndResponses/Employees/AddUpdateEmployeeRequest.cs
using EmployeeEntity = Employee.Performance.Evaluator.Core.Entities.Employee;

namespace Employee.Performance.Evaluator.Application.RequestsAndResponses.Employees;

public class AddUpdateEmployeeRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public DateTimeOffset BirthDate { get; set; }
    public int TeamId { get; set; }
    public bool IsTeamLead { get; set; }
    public int UserId { get; set; }
    public string? Avatar { get; set; }

    public int RoleId { get; set; }

    public static EmployeeEntity ToDbModel(AddUpdateEmployeeRequest request)
    {
        return new EmployeeEntity
        {
            FirstName = request.FirstName,
            LastName = request.LastName,
            PhoneNumber = request.PhoneNumber,
            BirthDate = request.BirthDate,
            TeamId = request.TeamId,
            UserId = request.UserId,
            Avatar = string.IsNullOrEmpty(request.Avatar)
                ? null
                : Convert.FromBase64String(
                    request.Avatar.Split(',').Last()
                )
        };
    }
}

File: frontend/src/app/models/requests/add-update-employee.request.ts
import { Employee } from '../employee.model';

export interface AddUpdateEmployeeRequest {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  birthDate: string;
  teamId: number;
  isTeamLead?: boolean;
  userId: number;
  avatar: string | null;
  roleId: number;
}

export function mapToAddUpdateEmployeeRequest(
  employee: Employee
): AddUpdateEmployeeRequest {
  return {
    firstName: employee.firstName,
    lastName: employee.lastName,
    phoneNumber: employee.phoneNumber,
    birthDate: employee.birthDate,
    teamId: employee.teamId,
    userId: employee.userId,
    avatar: employee.avatar || null,
    roleId: employee.user.roleId,
  };
}
